python/py-configparser/Portfile

```diff
@@ -5,37 +5,49 @@ PortGroup           python 1.0
 
 name                py-configparser
 epoch               1
-version             4.0.2
+version             5.0.2
 revision            0
+
 categories-append   devel
 platforms           darwin
 license             MIT
 supported_archs     noarch
 
-python.versions     27
+homepage            https://github.com/jaraco/configparser
+
+description         Configparser from Python 3.8+ to Python 2.6-3.7
+long_description    ${description}
 
 maintainers         {stromnov @stromnov} openmaintainer
 
-description         Configparser from Python 3.8 to Python 2.6-3.7
-long_description    ${description}
+python.versions     27 39
+
+if {${name} ne ${subport}} {
 
-homepage            https://github.com/jaraco/configparser/
-master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+    # 4.0.2 is the last py27-compatible version
+    if {${python.version} == 27} {
+        version             4.0.2
 
-distname            ${python.rootname}-${version}
+        depends_lib-append  port:py${python.version}-backports
 
-checksums           rmd160  9b6a5b12490e72c47ce2b1d190a35687c2d2c7eb \
-                    sha256  c7d282687a5308319bf3d2e7706e575c635b0a470342641c93bea0ea3b5331df \
-                    size    72498
+        distname            ${python.rootname}-${version}
 
-# 4.0.2 is the last py27-compatible version
-livecheck.type      none
+        checksums           ${distname}${extract.suffix} \
```

Comment:
I don't think this `${distname}${extract.suffix}` is necessary here (and below) - that should be the default right, and there is only one file/checksum combination used

Reply:
just out of curiosity, why do you need this for PY39? Isn't this supposed to be backporting the implementation in the latest Python release to earlier versions?

Reply:
> I don't think this `${distname}${extract.suffix}` is necessary here (and below) - that should be the default right, and there is only one file/checksum combination used

Resolved

> just out of curiosity, why do you need this for PY39? Isn't this supposed to be backporting the implementation in the latest Python release to earlier versions?

I'm trying to port [pyinfra](https://pyinfra.com/), and it even for Python 3.9, it requires `configparser`.

Reply:
I suspect upstream might just have forgotten to add the correct Python version specifiers in the dependencies. Did you try to patch the setup.py and/or build without adding this dependency in the Portfile and see if it works?

Reply:
There is no harm in having this as a package available for Python 3.9

As a matter of fact, the 5.0.x version of this package is aimed at Python 3+, as you can see from `configparser`'s changelog: https://github.com/jaraco/configparser/blob/main/CHANGES.rst